src/bridge/MigrationNFT.sol

```diff
@@ -3,38 +3,51 @@
 pragma solidity 0.8.23;
 
 import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Strings} from "openzeppelin-contracts/contracts/utils/Strings.sol";
 import {NODLMigration} from "./NODLMigration.sol";
 
 contract MigrationNFT is ERC721 {
+    using Strings for uint256;
+
     uint256 public nextTokenId;
-    uint256 public maxNFTs;
-    uint256 public minAmount;
+    uint256 public maxHolders;
     NODLMigration public migration;
 
-    string internal tokensURI;
+    string internal tokensURIRoot;
+
+    uint256[] public levels;
```

Comment:
```suggestion
    string internal immutable tokensURIRoot;

    uint256[] public immutable levels;
```

Reply:
FYI @aliXsed you cannot make `uint256[]` and `string` immutable. `solc` will complain via `Error (6377): Immutable variables cannot have a non-value type.`